crawler.py

```diff
@@ -54,32 +53,31 @@ def filtered_out_links(links, domain_name):
     return filtered_links
 
 
-def updated_urls_list(urls, links):
-    """Upade urls list with new links."""
-    for link in links:
-        if link not in urls:
-            urls.append(link)
-    return urls
+def site_dictionary(urls, domain_name, output_dict=dict()):
+    """Returns dictionary which will return page title and links of page from given domain."""
+    if urls:
+        new_urls = []
+        for url in urls:
+            if url not in output_dict:
+                browser = mechanize.Browser()
+                browser.open(url)
+                all_links = page_links(browser)
+                domain_links = filtered_out_links(all_links, domain_name)
```

Comment:
I think that this is not necessary, but I did that because example from pdf had 'set()' when there was no links in page. I wanted to be as close to example solution as possible.
Should I change it to list?

Reply:
No, I understand now. It is fine.